pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -274,6 +280,10 @@ func (a *AddVolumeOptions) Validate(isAddOp bool) error {
 				if len(a.SecretName) == 0 {
 					return errors.New("must provide --secret-name for --type=secret")
 				}
+			case "configmap":
```

Comment:
> I have a feeling you'll prefer to defer this for a follow-up, but this validation logic is really hard to make sense of.

Yeah, tightening validation and fixing the method overall is a TODO kind of item.